treeherder/webapp/api/intermittents_view.py

```diff
@@ -0,0 +1,124 @@
+                                            bug_count=Count('job_id')).values('bug_id', 'bug_count').order_by(
+                                            '-bug_count')
+        return queryset
+
+
+class FailuresByBug(generics.ListAPIView):
+    """ List of intermittent failure job details by bug, date range and tree (project name) """
+
+    serializer_class = FailuresByBugSerializer
+    pagination_class = CustomPagePagination
+
+    def get_queryset(self):
+        startday = self.request.query_params.get('startday').encode('utf-8') + ' 00:00:00'
+        endday = get_end_of_day(self.request.query_params.get('endday').encode('utf-8'))
+        tree = list(get_tree(self.request.query_params.get('tree').encode('utf-8')))
+        bug_id = int(self.request.query_params.get('bug'))
+
+        queryset = BugJobMap.objects.filter(bug_id=bug_id,
+                                            job__repository_id__in=tree,
+                                            job__push__time__range=(startday, endday)
+                                            ).select_related('job', 'push').values(
+                                            'bug_id', 'job_id', 'job__push__time', 'job__machine_platform__platform',
+                                            'job__repository__name', 'job__push__revision',
+                                            'job__signature__job_type_name', 'job__option_collection_hash',
+                                             ).order_by('-job__push__time')
+
+        hash_list = []
+
+        for item in queryset:
+            test_type = item['job__signature__job_type_name']
+            item['test_suite'] = re.sub(r'.+/', '', test_type)
+            match = filter(lambda x: item['job__option_collection_hash'] == x, hash_list)
+            if len(match) == 0:
+                hash_list.append(item['job__option_collection_hash'])
+
+        hash_query = OptionCollection.objects.filter(option_collection_hash__in=hash_list).select_related(
+                                                     'option').values('option__name', 'option_collection_hash')
+
+        for item in queryset:
+            match = filter(lambda x: item['job__option_collection_hash'] == x['option_collection_hash'], hash_query)
+            if len(match) > 0:
+                item['build_type'] = match[0]['option__name']
+            else:
+                item['build_type'] = 'unknown'
+
+        return queryset
+
+
+class FailureCount(generics.ListAPIView):
+    """ List of failures (optionally by bug) and testruns by day per date range and tree"""
+
+    serializer_class = FailureCount
+
+    def get_queryset(self):
+        startday = self.request.query_params.get('startday').encode('utf-8') + ' 00:00:00'
+        endday = get_end_of_day(self.request.query_params.get('endday').encode('utf-8'))
+        tree = list(get_tree(self.request.query_params.get('tree').encode('utf-8')))
```

Comment:
@edmorley I'm not sure if you saw my other comment, but the reason why the query doesn't have a list of names to be filtered by repo__name is because anything more than 2 items in a list with the join results in max query limit timeouts at 60,000ms. However, searching by ID's on repo - regardless of the size of the list - is quick. I probably should've been a bit clearer about why I didn't go for the obvious route to begin with (even doing the raw sql in the dbshell it takes 30 sec to return results when searching by a list of 4). I tried it and couldn't even get the api's to work for a demo. :)

Reply:
Under the hood exactly the same type of query will be run, so there might have just been a typo or something causing that? Using django-debug-toolbar on a shorter range will show the query it's trying to run and make the cause clearer:
https://treeherder.readthedocs.io/common_tasks.html#profiling-api-endpoint-performance

Reply:
Current Implementation

![screen shot 2018-03-01 at 12 36 40 pm](https://user-images.githubusercontent.com/19615783/36868457-3b64a26e-1d4d-11e8-81a2-258813e2710e.png)

* the get_tree id lookup result is evaluated using list() before it hits the query, like so:
tree = list(get_tree(self.request.query_params.get('tree').encode('utf-8'))) 
* query is: BugJobMap.objects.filter(job__repository_id__in=(1, 2, 77, 14)...

Reply:
@edmorley 
![screen shot 2018-03-01 at 12 27 06 pm](https://user-images.githubusercontent.com/19615783/36868583-9fa767e8-1d4d-11e8-92d5-cc093dbdfa92.png)
*with query like so: 
queryset = BugJobMap.objects.filter(job__repository__name__in=['mozilla-central', 'mozilla-inbound', 'autoland', 'fx-team'] ...